### ISSUE TITLE
pra/x264/x264lib.c:154: error: ‘AVPacket’ undeclared (first use in this function)

### ISSUE THREAD

**Issue by totaam** [Xpra-org/xpra#100]
*Issue migrated from trac ticket # 100*

**component:** android | **priority:** major | **resolution:** wontfix

#### 2012-04-03 12:05:29: lindi created the issue

___
> building xpra from svn fails today:
> 
> ```
> skipping 'xpra/wait_for_x_server.c' Cython extension (up-to-date)
> building 'xpra.wait_for_x_server' extension
> creating build/temp.linux-x86_64-2.6/xpra
> gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -O0 -fPIC -I/usr/include/python2.6 -c xpra/wait_for_x_server.c -o build/temp.linux-x86_64-2.6/xpra/wait_for_x_server.o
> gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -O0 build/temp.linux-x86_64-2.6/xpra/wait_for_x_server.o -lX11 -o build/lib.linux-x86_64-2.6/xpra/wait_for_x_server.so
> cythoning xpra/x264/codec.pyx to xpra/x264/codec.c
> building 'xpra.x264.codec' extension
> creating build/temp.linux-x86_64-2.6/xpra/x264
> gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -O0 -fPIC -I/usr/include/python2.6 -c xpra/x264/codec.c -o build/temp.linux-x86_64-2.6/xpra/x264/codec.o
> xpra/x264/codec.c: In function ‘__pyx_pf_4xpra_4x264_5codec_7Decoder_2decompress_image’:
> xpra/x264/codec.c:1074: warning: passing argument 2 of ‘PyObject_AsReadBuffer’ from incompatible pointer type
> /usr/include/python2.6/abstract.h:501: note: expected ‘const void **’ but argument is of type ‘void **’
> xpra/x264/codec.c: In function ‘__pyx_pf_4xpra_4x264_5codec_7Encoder_2compress_image’:
> xpra/x264/codec.c:1498: warning: passing argument 2 of ‘PyObject_AsReadBuffer’ from incompatible pointer type
> /usr/include/python2.6/abstract.h:501: note: expected ‘const void **’ but argument is of type ‘void **’
> gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -O0 -fPIC -I/usr/include/python2.6 -c xpra/x264/x264lib.c -o build/temp.linux-x86_64-2.6/xpra/x264/x264lib.o
> xpra/x264/x264lib.c: In function ‘init_decoder’:
> xpra/x264/x264lib.c:91: warning: implicit declaration of function ‘avcodec_alloc_context3’
> xpra/x264/x264lib.c:91: warning: assignment makes pointer from integer without a cast
> xpra/x264/x264lib.c: In function ‘compress_image’:
> xpra/x264/x264lib.c:121: warning: passing argument 2 of ‘sws_scale’ from incompatible pointer type
> /usr/include/libswscale/swscale.h:155: note: expected ‘uint8_t **’ but argument is of type ‘const uint8_t **’
> xpra/x264/x264lib.c: In function ‘decompress_image’:
> xpra/x264/x264lib.c:154: error: ‘AVPacket’ undeclared (first use in this function)
> xpra/x264/x264lib.c:154: error: (Each undeclared identifier is reported only once
> xpra/x264/x264lib.c:154: error: for each function it appears in.)
> xpra/x264/x264lib.c:154: error: expected ‘;’ before ‘avpkt’
> xpra/x264/x264lib.c:160: warning: implicit declaration of function ‘av_init_packet’
> xpra/x264/x264lib.c:160: error: ‘avpkt’ undeclared (first use in this function)
> xpra/x264/x264lib.c:170: warning: implicit declaration of function ‘avcodec_decode_video2’
> error: command 'gcc' failed with exit status 1
> 
> ```
> 
> This is svn 693.
> 
> debian squeeze with libvpx-dev and libx264-dev from squeeze-backports.
> 
> amd64.
> 
>
